### ISSUE TITLE
should dispatch changeEvent after value is updated

### ISSUE THREAD

**Issue by hwqing** [furcan/KioskBoard#11]
Dear,

There is an issue that in change event for INPUT, you always get the old value.
For example, if currently the value for INPUT is "abc", and then click the VK key "d", you got "abcd" in INPUT, and in the meantime, change event triggered for INPUT and in this event, if you try to get the value of INPUT, you still got "abc".

By going through your code, I found that "input.dispatchEvent(changeEvent);" was executed before "input.value = theInputValArray.join('');" which might cause this problem.

Please consider to make relevant modification.
Thank you very much.

**Comment by furcan** [comment]
Hi @hwqing 

Thank you for using Kioskboard.js.

I did not get the issue or your expectation clearly.

But you can find an example video below to see everything works as normal. Otherwise, please explain your issue in more detail and with examples.

Thank you. 

Furkan.


https://user-images.githubusercontent.com/49978011/107542329-218e9600-6bd9-11eb-9489-cf4ae7819253.mp4

**Comment by gabe-lucas** [comment]
Hey @furcan,

I just wanted to start off by saying fantastic job on Kioskboard.
My goal is to give you a little bit of more clarification on what the issue is as I have encountered it as well.

When you're listening to the change event in JS, for example:
```
document.getElementById("example").addEventListener("change", (e) => {
    console.log(document.getElementById("example").value);
});```
This will print the value of the statement before it is updated by Kioskboard.

My solution to this was moving the event dispatch calls after setting the input values.
Let me know if you have any questions.


Kind Regards!

**Comment by furcan** [comment]
Hi @gabe-lucas , @hwqing 

The issue has been fixed (v1.4.0)

Thanks,
Furkan.
